devcycle_python_sdk/api/config_client.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import math
+import random
+import time
+from os.path import join
+from typing import Optional
+from http import HTTPStatus
+
+import requests
+
+from devcycle_python_sdk.dvc_options import DevCycleLocalOptions
+from devcycle_python_sdk.exceptions import (
+    CloudClientError,
+    NotFoundError,
+    CloudClientUnauthorizedError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigAPIClient:
+    def __init__(self, sdk_key: str, options: DevCycleLocalOptions):
+        self.sdk_key = sdk_key
+        self.options = options
+        self.session = requests.Session()
+        self.session.headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+        }
+        self.session.max_redirects = 0
+        self.max_config_retries = 2
+
+    def _config_file_url(self) -> str:
+        return join(self.options.config_CDN_URI, "v1", "server", self.sdk_key, ".json")
+
+    def get_config(self, config_etag: str = None) -> (dict, str):
+        retries_remaining = self.max_config_retries
+        timeout = self.options.config_request_timeout_ms / 1000.0
+
+        url = self._config_file_url()
+
+        headers = {}
+        if config_etag:
+            headers["If-None-Match"] = config_etag
+
+        attempts = 1
+        while retries_remaining > 0:
+            request_error: Optional[Exception] = None
+            try:
+                res: requests.Response = self.session.request(
+                    "GET", url, params={}, timeout=timeout, headers=headers
+                )
+
+                if res.status_code == HTTPStatus.UNAUTHORIZED or res.status_code == HTTPStatus.FORBIDDEN:
+                    # Not a retryable error
+                    raise CloudClientUnauthorizedError("Invalid SDK Key")
+                elif res.status_code == HTTPStatus.NOT_MODIFIED:
+                    # the config hasn't changed since the last request
+                    # don't return anything
+                    return None, config_etag
```

Comment:
I acknowledge that this is a bit odd.  I'm open to other ways to handle this situation better and report back to the caller that the data hasn't changed an no action should be taken.